hera_mc/cm_dataview.py

```
@@ -15,85 +15,161 @@
+                elif output == 'corr':
+                    if self.fc_map[p]['fc'][i]:
+                        this_hu = self.fc_map[p]['fc'][i][0].hookup
+                        c = hu.get_correlator_input_from_hookup(this_hu)
+                        s = 'e:{} n:{}'.format(c['e'], c['n'])
+                    else:
+                        s = '---'
+                    fp.write('{:^19}'.format(s))
+            fp.write('\n')
+        fp.close()
+
+    def read_fc_map_files(self, filenames):
+        """
+        Reads in fc_map from the supplied filenames list.
+
+        Parameters:
+        ------------
+        filenames:  list of filenames to read in .
+        """
+
+        fc_map = {}
+        parts = []
+        for fn in filenames:
+            try:
+                fp = open(fn, 'r')
+            except IOError:
+                print('{} not found.'.format(fn))
+                continue
+            print("Reading {}".format(fn))
+            part_fn = fp.readline().split()
+            del(part_fn[0])
+            for p in part_fn:
+                if p not in fc_map.keys():
+                    fc_map[p] = {'datetime': [], 'flag': [], 'fc': []}
+                    parts.append(p)
+            for line in fp:
+                data = line.split()
+                date = cm_utils._get_astropytime(data[0], data[1])
+                del(data[0:2])
+                for i in range(len(data)):
+                    fc_map[part_fn[i]]['datetime'].append(date.datetime)
+                    fc_map[part_fn[i]]['flag'].append(int(data[i]))
+        self.parts_list = parts
+        self.fc_map = fc_map
+        return parts, fc_map
+
+    def plot_fc_map(self):
+        """
+        Plots the fc_map flgs.
+
+        Parameters:
+        ------------
+        separate:  T/F to plot the rows separated.
+        """
+        if self.fc_map is None or self.parts_list is None:
+            print("You first need to generate fc_map and parts_list")
```

This should raise an exception.